extensions/gopass/src/details.tsx

```diff
@@ -0,0 +1,58 @@
+import { Action, ActionPanel, Clipboard, closeMainWindow, Icon, List, showHUD, showToast, Toast } from "@raycast/api";
+import { useEffect, useState } from "react";
+import gopass from "./gopass";
+import { humanize } from "./utils";
+import { copyPassword, pastePassword } from "./index";
+
+async function copy(key: string, value: string): Promise<void> {
+  Clipboard.copy(value);
+  await closeMainWindow();
+  await showHUD(`${key} copied`);
+}
+
+async function paste(key: string, value: string): Promise<void> {
+  await showHUD(`${key} pasted`);
+  await Clipboard.paste(value);
+  await closeMainWindow();
+}
+
+export default function ({ entry }: { entry: string }): JSX.Element {
+  const [details, setDetails] = useState<string[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect((): void => {
+    gopass
+      .show(entry)
+      .then((data) => ["password: *****************"].concat(data))
+      .then(setDetails)
+      .catch(async (error) => {
+        console.error(error);
+        await showToast({ title: "Could not load passwords", style: Toast.Style.Failure });
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  return (
+    <List isLoading={loading}>
+      {details.map((item, index) => {
+        const [key, value] = item.split(": ");
+        const copyAction = () => (index === 0 ? copyPassword(entry) : copy(key, value));
+        const pasteAction = () => (index === 0 ? pastePassword(entry) : paste(key, value));
```

Comment:
Do you think this would be more clear if you'd create the first `List.Item` manually outside the map? We could get rid of these the `concat` in line 26 and these two actions. We also wouldn't have to compare to an index.